File: src/CoreProject1/wwwroot/js/tripsController.js
(function() {
    'use strict';

    angular.module("app-trips")
        .controller("tripsController", tripsController);

    function tripsController($http) {
        var vm = this;
        vm.name = "test";
        vm.trips = [
            {
                name: "US Trip",
                created: new Date()
            }, {
                name: "World Trip",
                created: new Date()
            }
        ];

        vm.newTrip = {};
        vm.errorMessage = "";
        vm.isBusy = true;

        //-- Retrieving data
        $http.get("/api/trips")
            .then(function(response) {
                //success
                angular.copy(response.data, vm.trips);
                },
                function(error) {
                    //failure
                    vm.errorMessage = "Failed to load data: " + error;
                })
        .finally(function() {
                vm.isBusy = false;
        });

        //-- Creating data
        vm.addTrip = function() {
            vm.isBusy = true;
            vm.errorMessage = "";
            $http.post("/api/trips", vm.newTrip)
                .then(function(response) {
                    //-- success
                        //alert("added");
                    vm.trips.push(response.data);
                        vm.newTrip = {};
                    },
                    function() {
                        //-- error
                        vm.errorMessage = "Failed to save new trip";
                    })
                .finally(function() {
                    vm.isBusy = false;
                });
        };

    }


})();

File: src/CoreProject1/Controllers/Api/TripsController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using AutoMapper;
using CoreProject1.Models;
using CoreProject1.ViewModels;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore.Storage;
using Microsoft.Extensions.Logging;

namespace CoreProject1.Controllers.Api
{    [Authorize]
    [Route("api/trips")]
    public class TripsController: Controller
    {
        private IWorldRepository _repository;
        private ILogger<TripsController> _logger;

        public TripsController(IWorldRepository repository, ILogger<TripsController> logger)
        {
            _repository = repository;
            _logger = logger;

        }

        [HttpGet("")]
        public IActionResult Get()
        {
            try
            {
                var results = _repository.GetTripByUsername(this.User.Identity.Name);
                return Ok(Mapper.Map<IEnumerable<TripViewModel>>(results));
            }
            catch (Exception ex)
            {
                //--TODO Logging
                _logger.LogError($"Failed to get All Trips: {ex}");

                return BadRequest(ex);
            }
        }

        [HttpPost("")]
        public async Task<IActionResult> Post([FromBody]TripViewModel theTrip)
        {
            if (ModelState.IsValid)
            {
                var newTrip = Mapper.Map<Trip>(theTrip);

                newTrip.UserName = User.Identity.Name;
                _repository.AddTrip(newTrip);

                //Save to the database
                if (await _repository.SaveChangesAsync())
                {
                    return Created($"api/trips/{theTrip.Name}", Mapper.Map<TripViewModel>(newTrip));
                }
                else
                {
                    return BadRequest("Failed to save changes to the database");
                }
            }
            //return BadRequest("Bad data");
            return BadRequest(ModelState);
        }

        
        
    }
}
